### PR TITLE
Add in support for GetStructField

### PR DESCRIPTION
**Pull request by revans2**
Adds in basic support for getting a field from a struct.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala [hunk 1]
```diff
@@ -17,12 +17,57 @@
...
-import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, ExtractValue, GetArrayItem, GetMapValue, ImplicitCastInputTypes, NullIntolerant}
-import org.apache.spark.sql.catalyst.util.TypeUtils
-import org.apache.spark.sql.types.{AbstractDataType, AnyDataType, ArrayType, DataType, IntegralType, MapType, StringType, StructType}
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, ExtractValue, GetArrayItem, GetMapValue, ImplicitCastInputTypes, NullIntolerant, UnaryExpression}
+import org.apache.spark.sql.catalyst.util.{quoteIdentifier, TypeUtils}
+import org.apache.spark.sql.types.{AbstractDataType, AnyDataType, ArrayType, DataType, IntegralType, MapType, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class GpuGetStructField(child: Expression, ordinal: Int, name: Option[String] = None)
+    extends UnaryExpression with GpuExpression with ExtractValue {
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala:31]: Spark version has `with NullIntolerant`. Probably doesn't matter here, but I noticed we're applying it to many other expressions.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala [hunk 2]
```diff
@@ -17,12 +17,57 @@
...
+    val fieldName = if (resolved) childSchema(ordinal).name else s"_$ordinal"
+    s"$child.${name.getOrElse(fieldName)}"
+  }
+
+  override def sql: String =
+    child.sql + s".${quoteIdentifier(name.getOrElse(childSchema(ordinal).name))}"
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val input = child.columnarEval(batch)
+    val dt = dataType
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala:48]: Nit: This should be inside the `try` block.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala (hunk 1 above)
Comment (revans2) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala:31]: https://github.com/apache/spark/blob/3695e997d5d436be086235505bbb030c87ae8eef/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala#L368-L373

It is used as a part of planning only and does not really do anything for us, but will add it.

**Review by kuhushukla — DISMISSED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe @kuhushukla please take another look

**Review by kuhushukla — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Pull request merged**